src/Temporalio/Client/TemporalClient.Workflow.cs

```diff
@@ -115,6 +155,161 @@ public override async Task<WorkflowHandle<TWorkflow, TResult>> StartWorkflowAsyn
+                    {
+                        throw new ArgumentException("Start workflow operation is required in options");
+                    }
+                    if (input.Options.StartWorkflowOperation.Options.StartSignal != null ||
+                        input.Options.StartWorkflowOperation.Options.StartSignalArgs != null)
+                    {
+                        throw new ArgumentException("Cannot have start signal on update with start");
+                    }
+                    if (input.Options.StartWorkflowOperation.Options.RequestEagerStart)
+                    {
+                        throw new ArgumentException("Cannot request eager start on update with start");
+                    }
+                    if (input.Options.StartWorkflowOperation.Options.IdConflictPolicy == WorkflowIdConflictPolicy.Unspecified)
+                    {
+                        throw new ArgumentException("Workflow ID conflict policy required for update with start");
+                    }
+                    if (input.Options.StartWorkflowOperation.Options.Rpc != null)
+                    {
+                        throw new ArgumentException("Cannot set RPC options on start options, set them on the update options");
+                    }
+
+                    // Single atomic to prevent reuse
+                    if (!input.Options.StartWorkflowOperation.TryMarkUsed())
+                    {
+                        throw new ArgumentException("Start operation already used");
+                    }
+
+                    // Build request
+                    var startReq = await CreateStartWorkflowRequestAsync(
+                        input.Options.StartWorkflowOperation.Workflow,
+                        input.Options.StartWorkflowOperation.Args,
+                        input.Options.StartWorkflowOperation.Options,
+                        input.Options.StartWorkflowOperation.Headers).ConfigureAwait(false);
+                    var updateReq = await CreateUpdateWorkflowRequestAsync(
+                        input.Update,
+                        input.Args,
+                        input.Options,
+                        input.Options.WaitForStage,
+                        input.Headers).ConfigureAwait(false);
+                    updateReq.WorkflowExecution = new() { WorkflowId = startReq.WorkflowId };
+                    var req = new ExecuteMultiOperationRequest() { Namespace = Client.Options.Namespace };
+                    req.Operations.Add(
+                        new ExecuteMultiOperationRequest.Types.Operation() { StartWorkflow = startReq });
+                    req.Operations.Add(
+                        new ExecuteMultiOperationRequest.Types.Operation() { UpdateWorkflow = updateReq });
+
+                    // Continually try to start until an exception occurs, the user-asked stage is
+                    // reached, or the stage is accepted. But we will set the workflow handle as soon as
```

Comment:
See comment below, I think `while < Accepted` is all that's needed.

Reply:
Updated here and in regular update logic